src/k8s/pkg/k8sd/features/metallb/loadbalancer.go

```diff
@@ -0,0 +1,117 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/k8s/pkg/client/helm"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+	"github.com/canonical/k8s/pkg/utils/control"
+)
+
+func ApplyLoadBalancer(ctx context.Context, snap snap.Snap, loadbalancer types.LoadBalancer, network types.Network, _ types.Annotations) error {
+	if !loadbalancer.GetEnabled() {
+		if err := disableLoadBalancer(ctx, snap, network); err != nil {
+			return fmt.Errorf("failed to disable LoadBalancer: %w", err)
+		}
+		return nil
+	}
+
+	if err := enableLoadBalancer(ctx, snap, loadbalancer, network); err != nil {
+		return fmt.Errorf("failed to enable LoadBalancer: %w", err)
+	}
+	return nil
+}
+
+func disableLoadBalancer(ctx context.Context, snap snap.Snap, network types.Network) error {
+	m := snap.HelmClient()
+
+	if _, err := m.Apply(ctx, chartMetalLBLoadBalancer, helm.StateDeleted, nil); err != nil {
+		return fmt.Errorf("failed to uninstall MetalLB LoadBalancer chart: %w", err)
+	}
+
+	if _, err := m.Apply(ctx, chartMetalLB, helm.StateDeleted, nil); err != nil {
+		return fmt.Errorf("failed to uninstall MetalLB chart: %w", err)
+	}
+	return nil
+}
+
+func enableLoadBalancer(ctx context.Context, snap snap.Snap, loadbalancer types.LoadBalancer, network types.Network) error {
+	m := snap.HelmClient()
+
+	if _, err := m.Apply(ctx, chartMetalLB, helm.StatePresent, nil); err != nil {
+		return fmt.Errorf("failed to apply MetalLB configuration: %w", err)
+	}
+
+	if err := waitForRequiredLoadBalancerCRDs(ctx, snap, loadbalancer.GetBGPMode()); err != nil {
+		return fmt.Errorf("failed to wait for required MetalLB CRDs: %w", err)
+	}
+
+	cidrs := []string{}
+	cidrs = append(cidrs, *loadbalancer.CIDRs...)
+
+	for _, ipRange := range loadbalancer.GetIPRanges() {
+		cidrs = append(cidrs, fmt.Sprintf("%s-%s", ipRange.Start, ipRange.Stop))
+	}
```

Comment:
this is quite problematic for two reasons:
- we always dereference `loadbalancer.CIDRs`, so we'll panic if that field is nil
- we mix data of different schemas in the same place. this is something that will be very confusing for future debugging efforts, and how this is handled in the chart / code / tests etc.

Instead, please do the same thing we do elsewhere, https://github.com/canonical/k8s-snap/blob/f026b18c18a4e80d5dd0b6147ef6a638127d415d/src/k8s/pkg/k8sd/features/cilium/loadbalancer.go#L93-L99